inc/RequestParser.hpp

```diff
@@ -20,12 +20,15 @@
 
 /* ====== DEFINITIONS ====== */
 
-const int decimalBase = 10;
+const int g_decimalBase = 10;
 
 /* ====== CLASS DECLARATION ====== */
 
 class RequestParser {
 public:
+	static const int s_maxLabelLength = 63; /**< Maximum length for labels (the parts between dots in a domain name)  */
+	static const int s_maxHostNameLength = 255; /**< Maximum length for hostname */
```

Comment:
I think a webserver would mostly deal with hostnames used in DNS. Therefore I would limit to 253

From [Stackoverflow](https://stackoverflow.com/questions/8724954/what-is-the-maximum-number-of-characters-for-a-host-name-in-unix/28918017#28918017): "Hostnames used in DNS can be as long as 253 bytes as a fully qualified domain name" 

Or the linked [MSDN blog article ](https://devblogs.microsoft.com/oldnewthing/20120412-00/?p=7873) with useful illustration 
(in here it's also mentioned that with utf-8 we would need to calculate differently - let's just ignore that :D )

Reply:
Really nice article, very neatly explained. Changed to 253!